spec/adapter/base_spec.cr

```diff
@@ -1,32 +1,40 @@
 require "../spec_helper"
 
+Granite::Settings.adapters << Granite::Adapter::Pg.new({name: "FooBar123", url: "FAKE_DB"})
+
+class Foo < Granite::Base
+  adapter FooBar123
+end
+
 describe Granite::Adapter::Base do
-  describe "#env" do
-    context "no ${} exists" do
-      it "should return itself" do
-        url = "dummy://user:pswd@host:3333/database"
-        Granite::Adapter::Base.env(url).should eq url
+  describe "registration" do
```

Comment:
Since the tests in this file mess with the global scope of tests, it has some potential to create problems with other tests when run in different orders. Is it possible to use some test interface to run these tests without depending on the global scope of Granite::Settings?

Reply:
Possibly?  Really i could probably refactor that to use a one of the predefined classes and just test it based off of adapters already registered.  

i.e.

```Crystal
class Foo
  adapter mysql
end

adapter = Foo.adapter
adapter.name.should eq "mysq"
adapter.url.should eq ENV['MYSQL_DATABASE_URL"]
```

Would that be better?